shared/modules/transaction.utils.ts

```diff
@@ -54,6 +54,8 @@ type InferTransactionTypeResult = {
   getCodeResponse: string | null | undefined;
 };
 
+type DataMessageParam = object | string | number | boolean | JsonRpcParams;
```

Comment:
The only supported types here should be `string` and `object` I believe.

Reply:
Actually it complains the type because the rpc params can be:
```
(property) params: ((({
    [x: string]: Json;
} | (string | number | boolean | Json[] | {
    [prop: string]: Json;
} | null)[]) & {
    _exactOptionalGuard?: unique symbol;
}) | undefined) & (((Json[] | Record<string, Json>) | undefined) & [unknown, string | object])
```

I could force a cast to `string | object` before pass it to the `parseTypedDataMessage` function.